demos-europe/demosplan-core
# F dplan 11379 add customer to news

### Ticket
https://demoseurope.youtrack.cloud/issue/DPLAN-11379/ADO-Issue-15401-Portal-Mitteilungen-nur-im-Mandanten-sichtbar-in-dem-sie-erstellt-wurden

- Global news are not linked to any specific customer.
- In this PR, each global news item from _platform_content will be duplicated for every customer. This allows each customer to choose whether to keep or delete the news.
- Three tables are involved:
  - _platform_content: Stores the news.
  - _platform_content_categories: Links global news to a category.
  - _manually_list_sort: Stores the order of the news.
- The news order will not be included in the migration. News will be copied to customers without retaining the original order.

### How to review/test
- First go to  diplanbau project and then /news/verwalten
- Check that there are those news
- Go to your DB and check the table _platform_content (there is not customer id)
- Then Run the migration bin <project> dp:migrate
- Then check the table _platform_content, you will see the customer id column and another entries (as the existing entries where duplicated per customer)
- Refresh the page /news/verwalten
- You will see the news have a different order (that is ok) they should be sorted by created_at
- Then Edit - delete the news (all should work)


Delete the checkbox if it doesn't apply/isn't necessary.

- [ x] Move the tickets on the board accordingly
